pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -184,6 +184,14 @@ public static List<String> extractTableNamesFromNode(SqlNode sqlNode) {
       }
       tableNames.addAll(extractTableNamesFromNode(((SqlWith) sqlNode).body));
       tableNames.removeAll(aliases);
+    } else if (sqlNode instanceof SqlSetOption) {
+      for (SqlNode node : ((SqlSetOption) sqlNode).getOperandList()) {
+        if (node instanceof SqlIdentifier) {
+          tableNames.addAll(((SqlIdentifier) node).names);
```

Comment:
did we check we will produce identifier here (as table identifier)? or this is a simple copy paste?

Reply:
It's actually won't produce SqlIdentifier.